papaya-web/src/main/java/pl/edu/agh/papaya/rest/projects/controller/ProjectsController.java

```diff
@@ -98,4 +103,53 @@ private Project getValidProject(Long id) {
 
         return ResponseEntity.ok(sprints);
     }
+
+    @Override
+    public ResponseEntity<UserAvailabilityDto> getUserAvailability(Long projectId, Long sprintId) {
+        Project project = getValidProject(projectId);
+        Sprint sprint = getValidSprint(sprintId);
+
+        if (!isSprintInProject(sprint, project)) {
+            throw new ForbiddenAccessException();
+        }
+
+        Availability availability = availabilityService
+                .findBySprintIdAndUserId(sprintId, userContext.getUserIdn())
+                .orElseThrow(ResourceNotFoundException::new);
+
+        return ResponseEntity.ok(availabilityMapper.mapToApi(availability));
+    }
+
+    @Override
+    public ResponseEntity<UserAvailabilityDto> updateUserAvailability(@Valid UserAvailabilityDto body, Long projectId,
+            Long sprintId) {
+        Project project = getValidProject(projectId);
+        Sprint sprint = getValidSprint(sprintId);
+        if (!isSprintInProject(sprint, project) || !isAvailabilityDeclarationAllowed(sprint)) {
+            throw new ForbiddenAccessException();
+        }
+
+        Availability availability = availabilityService
+                .findBySprintIdAndUserId(sprintId, userContext.getUserIdn()).orElseGet(Availability::new);
+
+        availability.setNotes(body.getNotes());
+        availability.setPastSprintRemainingTime(Duration.ofMinutes(body.getTimeRemaining()));
```

Comment:
To nie chcemy nulla? Jak komuś nic nie zostało, to przyleci 0? Tak się czepiam, bo ja gdzieś sobie przegapiłam ostatnio, że pole może być nullem

Reply:
W sumie nie wiem, czy required w yamlu nie rozwiązuje takiego problemu, sprawdzę

Reply:
Nie jestem pewny ale required w yamlu (tak jak i w adnotacjach) działa chyba pasywnie, czyli tylko w swagger UI się wyświetli czy required czy nie.

Reply:
nie, przy podaniu nulla dla pola required, zwraca Bad Request